packages/styled/components/hoc/withThemes.js

```diff
@@ -1,18 +1,34 @@
-import React from 'react';
+import React, { useContext } from 'react';
 import PropTypes from 'prop-types';
 import getDisplayName from '@irvingjs/core/utils/getDisplayName';
 import assign from 'lodash/fp/assign';
+import get from 'lodash/get';
+import { ThemeContext } from 'styled-components';
 
 const withThemes = (themeMap) => (WrappedComponent) => {
   const ThemedComponent = (props) => {
-    const { themeName } = props;
+    const { themeName, style } = props;
     const theme = assign(
       themeMap.default,
       themeMap[themeName]
     );
 
+    // If the value of a property in a styled components `style` property is
+    // available in the styled component theme, use that value. This allows us
+    // to do things like `color: theme.colors.black` and have it become
+    // `color: #000000`.
+    const themeContext = useContext(ThemeContext);
+    const newProps = props;
+    Object.keys(style).forEach((property) => {
+      newProps.style[property] = get(
+        themeContext,
+        newProps.style[property],
+        newProps.style[property]
+      );
+    });
```

Comment:
A reducer makes it more clear what the themeContext will replace in the style prop.

```suggestion
const newStyle = Object.keys(style)
  .reduce((acc, property) => {
    const themeStyles = get(
      themeContext,
      style[property],
      style[property] // is this right, should the key be the default value?
    );
    
    return {
      ...acc,
      [property]: themeStyles,
    };
  }, style);
const newProps = { ...props, style: newStyle };
```

Reply:
Thanks, this makes sense.